core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -139,6 +139,25 @@ public String name() {
     return catalogName;
   }
 
+  private boolean isTableDir(Path path) {
+    Path metadataPath = new Path(path, "metadata");
+    // Only the path which contains metadata is the path for table, otherwise it could be
+    // still a namespace.
+    try {
+      return fs.listStatus(metadataPath, TABLE_FILTER).length >= 1;
+    } catch (IOException e) {
+      return false;
+    }
+  }
+
+  private boolean isDirectory(Path path) {
+    try {
+      return fs.getFileStatus(path).isDirectory();
+    } catch (IOException e) {
+      return false;
```

Comment:
This should throw either `RuntimeIOException` or `UncheckedIOException`. If the status can't be loaded, returning false is not correct.